contract/lst-oracle/src/lib.rs

```diff
@@ -0,0 +1,177 @@
+#![allow(clippy::needless_pass_by_value)]
+
+use near_sdk::{
+    assert_one_yocto, borsh::BorshSerialize, collections::UnorderedMap, env, near,
+    serde::de::DeserializeOwned, serde_json, AccountId, BorshStorageKey, Gas, IntoStorageKey,
+    PanicOnDefault, Promise, PromiseResult,
+};
+use near_sdk_contract_tools::{owner::Owner, Owner};
+use templar_common::{
+    define_list,
+    number::Decimal,
+    oracle::{
+        price_transformer::PriceTransformer,
+        pyth::{ext_pyth, OracleResponse, PriceIdentifier},
+    },
+    self_ext,
+};
+
+#[derive(BorshSerialize, BorshStorageKey)]
+#[borsh(crate = "near_sdk::borsh")]
+enum StorageKey {
+    Transformers,
+}
+
+#[derive(Debug, Owner, PanicOnDefault)]
+#[near(contract_state)]
+pub struct Contract {
+    pub oracle_id: AccountId,
+    pub transformers: UnorderedMap<PriceIdentifier, PriceTransformer>,
+}
+
+#[near]
+impl Contract {
+    #[init]
+    pub fn new(oracle_id: AccountId) -> Self {
+        let mut self_ = Self {
+            oracle_id,
+            transformers: UnorderedMap::new(StorageKey::Transformers.into_storage_key()),
+        };
+
+        Owner::init(&mut self_, &env::predecessor_account_id());
+
+        self_
+    }
+
+    pub fn get_oracle_id(&self) -> AccountId {
+        self.oracle_id.clone()
+    }
```

Comment:
Rename to `oracle_id` and its best to return always as a `&AccountId`, which later someone can choose to turn into an owned value later.